### PR TITLE
Refactored the component board render flow + fixed issues with eslint

### PR DESCRIPTION
**Pull request by fvaldez421**

### CODE REVIEW

File: client/.eslintrc.cjs [hunk 1]
```diff
@@ -1,11 +1,12 @@
 module.exports = {
   extends: ['mantine'],
   parserOptions: {
-    project: './tsconfig.json',
+    project: './client/tsconfig.json',
   },
-  workingDirectories: [{ mode: 'auto' }],
```
Comment (fvaldez421) [inline client/.eslintrc.cjs:6]: this throws some "top level configuration" issue bc its extending mantine

File: client/package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "private": true,
   "version": "1.0.0",
   "scripts": {
-    "dev": "node -r ./preload node_modules/.bin/vite",
+    "dev": "node -r ./preload ../node_modules/.bin/vite",
```
Comment (fvaldez421) [inline client/package.json:8]: yarn 4.1 doesnt install dependencies at the lower workspace level anymore...

**Comment by fvaldez421** [comment]
WARNING: Merging this will require all collaborators to upgrade to yarn@4.1.0

**Review by nad0m — APPROVED** [review]
Nice! I was actually thinking of cleaning up and abstracting out some of the common board game stuff for reusability too

**Pull request merged**